app/services/service_provider_updater.rb

```diff
@@ -0,0 +1,47 @@
+class ServiceProviderUpdater
+  def run
+    dashboard_service_providers.each do |sp|
+      issuer = sp['issuer'] || sp[:issuer]
+      SERVICE_PROVIDERS[issuer] = sp
+
+      VALID_SERVICE_PROVIDERS << issuer unless VALID_SERVICE_PROVIDERS.include?(issuer)
```

Comment:
What if we made `VALID_SERVICE_PROVIDERS` a set so we don't have to worry about double adding things?